src/analyses/goto_check.cpp

```diff
@@ -1017,14 +1017,14 @@ goto_checkt::address_check(const exprt &address, const exprt &size)
     if(flags.is_unknown() || flags.is_dynamic_heap())
     {
       conditions.push_back(conditiont(
-        not_exprt(deallocated(address, ns)),
+        or_exprt(allocs, not_exprt(deallocated(address, ns))),
```

Comment:
`allocs` seems like it should be a list of allocations, not a boolean, how about renaming it `in_bounds_of_some_alloc`?

Reply:
I've added a second commit doing the suggested renaming, thanks!